docs/DeveloperGuide.md

```diff
@@ -122,6 +149,63 @@ The following *sequence diagram* shows how the **delete property** operation wor
 
 ![Delete Property Sequence Diagram](diagrams/DeletePropertySD.png)
 
+### PairingList
+
+```PairingList``` facilities that pair and unpair commands by storing client-property pairs.
+
+When client rents a property, the client and property form a pair.
+
+*  ```PairingList``` uses a hash map to represent these client-property pairs, where the key is a ```Client``` object
+  and the value is a ```Property``` object.
+* A hash map is chosen due to its constant time lookup performance, making it efficient at querying the property that a
+  client is renting.
+* Also, the Java HashMap prevents duplicate keys, which dovetails nicely with the fact that real-life tenants only have
+  one place of residence at any time.
+
+#### Pair
+
+Here is a sequence diagram for the pair command, called from ```Duke.java```:
```

Comment:
I think this statement can be better phrase. For example, it can be phrased as "The sequence diagram for pair command is called from Duke.java as shown below: